ui-speedspacechart/src/components/common/DetailsBox.tsx

```diff
@@ -43,19 +43,19 @@ const DetailsBox = ({
   let rightOffset = 0;
   let bottomOffset = 0;
 
-  const adaptedOffset = store.isSettingsPanelOpened ? 641 : 115;
+  const detailsBoxWidth = document.getElementById('details-box')?.offsetWidth || 0;
+  const detailsBoxHeight = document.getElementById('details-box')?.offsetHeight || 0;
 
   // find out if the box is going out off the right side of the canvas
-  if (curveX + adaptedOffset > width - MARGIN_RIGHT - 10) rightOffset = 127;
+  if (curveX + detailsBoxWidth > width - MARGIN_RIGHT - 10) rightOffset = detailsBoxWidth + 12;
   // find out if the box is going out off the bottom side of the canvas
-  if (curveY + 180 > height - MARGIN_BOTTOM - 10) bottomOffset = 192;
+  if (curveY + detailsBoxHeight > height - MARGIN_BOTTOM - 10) bottomOffset = detailsBoxHeight + 12;
```

Comment:
Can we avoid having this magic number?

Reply:
I think you wasn't in the right branch, it's already wrote this way

Reply:
I know this file already has a lot of magic numbers. We should rewrite this in another PR.
I was talking of the `12` which is new.